sql/2024/cms/cms_adoption_by_rank.sql

```diff
@@ -0,0 +1,67 @@
+#standardSQL
+# CMS adoption per rank
+# cms_adoption_by_rank.sql
+
+SELECT
+  client,
+  technology,
+  rank,
+  COUNT(DISTINCT page) AS pages,
+  ANY_VALUE(total) AS total,
+  COUNT(DISTINCT page) / ANY_VALUE(total) AS pct
+FROM (
+  SELECT DISTINCT
+    client,
+    technology,
+    page
+  FROM
+    `httparchive.all.pages`,
+    UNNEST(technologies) AS technologies,
+    UNNEST(technologies.categories) AS cats
+  WHERE
+    date = '2024-06-01' AND
+    cats = 'CMS' AND
+    is_root_page
+)
+JOIN (
+  SELECT
+    client,
+    page,
+    rank
+  FROM
+    `httparchive.all.pages`
+  WHERE
+    date = '2024-06-01' AND
+    is_root_page
+)
+USING
+  (client,
+    page)
+JOIN (
+  SELECT
+    client,
+    rank_magnitude AS rank,
+    COUNT(0) AS total
+  FROM
+    `httparchive.all.pages`,
+    UNNEST([1e3, 1e4, 1e5, 1e6, 1e7]) AS rank_magnitude
+  WHERE
+    rank <= rank_magnitude AND
+    date = '2024-06-01' AND
+    is_root_page
+  GROUP BY
+    client,
+    rank_magnitude)
+USING
+  (client,
+    rank),
+  UNNEST([1e3, 1e4, 1e5, 1e6, 1e7]) AS rank_magnitude
```

Comment:
This query is incorrect as since 2022 we have also added ranks for 5000, 50,000...etc. Consequently, joining on `rank_magnitude` in L57 will not work since the `rank_magnitude` uses multiples of 1000.

Finally, I am not sure how this query is expected to differ from `top_cms_by_rank.sql`. I propose removing it.

Reply:
> This query is incorrect as since 2022 we have also added ranks for 5000, 50,000...etc. Consequently, joining on rank_magnitude in L57 will not work since the rank_magnitude uses multiples of 1000.

Should be fine as we just us `rank <= rank_magnitude`. So it's OK to skip some rankings (like the mid-steps). I think the mid-steps are more confusing and Web Almanac standard has been not to use them.

> Finally, I am not sure how this query is expected to differ from top_cms_by_rank.sql. I propose removing it.

That I don't know as not looked closely :-)

Reply:
Yes that is usually OK, but this query currently joins rank (5,000) with the rank_magnitude (10,000), so the join fails and misses some records. We would need to use a rank or rank magnitude for both sub-queries.

Reply:
Oh good spot! Yeah this is evil:

```sql
rank_magnitude AS rank,
```

Missed that totally! Should never reuse a real column name with something different IMHO.

Reply:
Agreed. I will remove it then. After having it fixed, the results are identical to `top_cms_by_rank.sql`

Reply:
Agreed. After fixing it, the results are identical to `top_cms_by_rank.sql` so I will proceed to remove it.